src/c/profile/transport/can/can_transport_posix.c

```diff
@@ -0,0 +1,138 @@
+#include <uxr/client/profile/transport/can/can_transport_posix.h>
+#include "can_transport_internal.h"
+
+#include <unistd.h>
+#include <string.h>
+#include <net/if.h>
+#include <sys/ioctl.h>
+
+bool uxr_init_can_platform(
+        uxrCANPlatform* platform,
+        const char* dev,
+        uint32_t can_id)
+{
+    static int enable_canfd = 1;
+    bool rv = false;
+
+    platform->poll_fd.fd = socket(PF_CAN, SOCK_RAW, CAN_RAW);
+
+    if (-1 != platform->poll_fd.fd)
+    {
+        struct sockaddr_can address;
+        struct ifreq ifr;
+
+        // Get interface index by name
+        strcpy(ifr.ifr_name, dev);
+        ioctl(platform->poll_fd.fd, SIOCGIFINDEX, &ifr);
+
+        memset(&address, 0, sizeof(address));
+        address.can_family = AF_CAN;
+        address.can_ifindex = ifr.ifr_ifindex;
+
+        if (-1 != bind(platform->poll_fd.fd,
+                (struct sockaddr*) &address,
+                sizeof(address)))
+        {
+            // Enable CAN FD
+            if (-1 != setsockopt(platform->poll_fd.fd, SOL_CAN_RAW, CAN_RAW_FD_FRAMES,
+                    &enable_canfd, sizeof(enable_canfd)))
+            {
+                platform->poll_fd.events = POLLIN;
+                platform->can_id = can_id | CAN_EFF_FLAG;
+
+                // Add CAN filter for micro-ROS data
+                struct can_filter rfilter;
+                rfilter.can_id = platform->can_id;
+                rfilter.can_mask = CAN_EFF_MASK;
+                setsockopt(platform->poll_fd.fd, SOL_CAN_RAW, CAN_RAW_FILTER, &rfilter, sizeof(rfilter));
+                rv = true;
+            }
+
+
+        }
+    }
+    return rv;
+}
+
+bool uxr_close_can_platform(
+        uxrCANPlatform* platform)
+{
+    return (-1 == platform->poll_fd.fd) ? true : (0 == close(platform->poll_fd.fd));
+}
+
+size_t uxr_write_can_data_platform(
+        uxrCANPlatform* platform,
+        const uint8_t* buf,
+        size_t len,
+        uint8_t* errcode)
+{
+    struct canfd_frame frame = {
+        .can_id = platform->can_id, .len = (uint8_t)len
```

Comment:
Where is guaranteed that this `len` cast is not going to overflow?

Reply:
I assumed that `len` will never be bigger than the transport MTU, which fits inside an `uint8_t`:
https://github.com/eProsima/Micro-XRCE-DDS-Client/blob/0f4019b01cc1851c92d7a9e4e19ec2ed6c7d4729/include/uxr/client/profile/transport/can/can_transport.h#L32